### DISCUSSION TITLE
How to manage a light with PIR sensor

### DISCUSSION THREAD

**Discussion by Baptiste-mrch** [arendst/Tasmota#17519] [Q&A]
Hello,

I am trying to make a light that activate for X second using a PIR sensor. I add a button for manual long-term activation, but I don't know how to manage these inputs. I make a truth table to represent what I'm trying to make:
![image](https://user-images.githubusercontent.com/62337736/209674749-7d9a367d-5d8e-47fe-be10-55c7c0decd2b.png)

Does anyone have a solution?
Thanks!

**Comment by barbudor** [comment]
Rules, RuleTimer, Switchmode
These are the key things you need to explore
https://tasmota.github.io/docs/Rules/

**Reply by sfromis** [reply]
This page also has suggestions about how you can handle a PIR
https://tasmota.github.io/docs/PIR-Motion-Sensors/

A general strategy for managing a tiny state machine could be to have the PIR on/off add/subtract 1 from a var, and have the button (probably another switch input) on/off add/subtract 2 from the same var. Then the var can have values from 0 to 3, and you can have rules to do what you need for each of those.
